src/qt/checklst.cpp

```diff
@@ -80,8 +80,13 @@ bool wxCheckListBox::IsChecked(unsigned int n) const
 
 void wxCheckListBox::Check(unsigned int n, bool check )
 {
+    // Prevent the emission of wxEVT_CHECKLISTBOX event by temporarily block all
+    // signals on m_qtListWidget object. QSignalBlocker can be used instead when
+    // Qt 5.3 becomes the minimum supprted version.
```

Comment:
FWIW we could easily have our own trivial RAII helper calling `blockSignals()` in ctor and unblocking them (if necessary) in dtor.

And use it not only here but in all the other places were we call `blockSignals()` manually currently.

Reply:
I just found this:
https://github.com/wxWidgets/wxWidgets/blob/5002fb0351d3e04a70d15e50f787d1589bf29053/src/qt/treectrl.cpp#L1119-L1137

I'll rename it to `wxQtSignalBlocker` and put it in `qt/private/winevent.h`
is that acceptable ?

Reply:
Yes, please. I was actually looking for this class because I thought it already existed, but didn't find it because I was looking in the headers and not in the sources.

TIA!

Reply:
P.S. Maybe it's not even necessary to rename it.